### PR TITLE
Centered images

### PR DESCRIPTION
**Pull request by bendardenne**
Introduces an option to place the configured image (`-i `option) centered on the screen (or all screens).

### CODE REVIEW

File: i3lock.c [hunk 1]
```diff
@@ -2195,9 +2206,10 @@ int main(int argc, char *argv[]) {
                 cairo_rectangle(ctx, 0, 0, last_resolution[0], last_resolution[1]);
                 cairo_fill(ctx);
                 cairo_pattern_destroy(pattern);
+            } else {
+                cairo_set_source_surface(ctx, img, 0, 0);
+                cairo_paint(ctx);
             }
-            cairo_set_source_surface(ctx, img, 0, 0);
-            cairo_paint(ctx);
```
Comment (Raymo111) [inline i3lock.c]: Originally if tile was set then `cairo_set_source_surface(ctx, img, 0, 0);` and `cairo_paint(ctx);` would occur, but now it wouldn't? Should we duplicate those two lines inside `else if (tile)` as well?

File: unlock_indicator.c [hunk 2]
```diff
@@ -1064,6 +1068,41 @@ void draw_image(uint32_t *resolution, xcb_drawable_t drawable) {
     cairo_destroy(xcb_ctx);
 }
 
+void draw_on_all_outputs(cairo_t* xcb_ctx) {
```
Comment (Raymo111) [inline unlock_indicator.c]: It'd be helpful if you could add some brief documentation describing what this does :)

File: i3lock.c (hunk 1 above)
Comment (Raymo111) [reply i3lock.c]: Nevermind I see this was mitigated in a later commit

File: unlock_indicator.c [hunk 3]
```diff
@@ -1064,14 +1052,64 @@ void draw_image(uint32_t *resolution, xcb_drawable_t drawable) {
     cairo_destroy(xcb_ctx);
 }
 
+void draw_image(uint32_t* resolution, cairo_t* xcb_ctx) {
```
Comment (Raymo111) [inline unlock_indicator.c:1047]: It'd be helpful if you could add some brief documentation describing the method :)

**Review by Raymo111 — CHANGES_REQUESTED** [review]
Yeah just add some documentation for that method and we should be good to go. Also my multi-monitor setup is really buggy (unrelated to i3lock) so if you could send a screencap of the center and tile options it would be much appreciated.

**Review by Raymo111 — APPROVED** [review]
Great! Looks good.

**Comment by Raymo111** [comment]
I'll merge this as soon as I see how it looks, just to be safe.

**Comment by bendardenne** [comment]
`./i3lock --no-verify  -C -i ~/.config/lock.png -c 00000088`  (centered)
Screenshot from 2020-04-19 01-15-30


`./i3lock --no-verify  -t -i ~/.config/lock.png -c 00000088`   (tiling)
Screenshot from 2020-04-19 01-15-55

` ./i3lock --no-verify -i ~/.config/lock.png -c 00000088 `    (neither)
Screenshot from 2020-04-19 01-17-44

**Comment by Raymo111** [comment]
Awesome! I approve of your use of Atom as well :)

**Comment by bendardenne** [comment]
> Awesome! I approve of your use of Atom as well :)

Well spotted! Right screen is indeed Atom, left one is CLion.

**Pull request merged**